pandas/core/groupby/ops.py

```diff
@@ -235,7 +235,7 @@ def size(self):
         if ngroup:
             out = np.bincount(ids[ids != -1], minlength=ngroup)
         else:
-            out = ids
+            out = []
```

Comment:
Hmm something just seems strange to me about this. So your test example works fine, but what happens if we have a DataFrame that has other groups besides the all-NA one? Can you add a test for that and make sure that still works?

Reply:
Done. In that case, the else block should never be used